src/main/java/net/dv8tion/jda/api/utils/data/DataArray.java

```diff
@@ -602,6 +603,25 @@ public DataArray remove(@Nullable Object value)
         return this;
     }
 
+    /**
+     * Serialize this object as JSON.
+     *
+     * @return a byte array containing the JSON representation of this object.
+     */
+    public byte[] toJson()
+    {
+        try
+        {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            mapper.writeValue(outputStream, data);
+            return outputStream.toByteArray();
+        }
+        catch (IOException e)
+        {
+            throw new IllegalStateException(e);
```

Comment:
I'd use `UncheckedIOException` instead even if it can't be thrown.

Reply:
Just copied the `toString()` method from right below, no big thoughts went into this. I don't understand the structure and failure handling of JDA good enough to make an educated decision anyways, feel free to edit at will.